idx/idx_gen.go

```diff
@@ -209,7 +243,141 @@ func (z *Archive) UnmarshalMsg(bts []byte) (o []byte, err error) {
 
 // Msgsize returns an upper bound estimate of the number of bytes occupied by the serialized message
 func (z *Archive) Msgsize() (s int) {
-	s = 1 + 17 + z.MetricDefinition.Msgsize() + 9 + msgp.Uint16Size + 6 + msgp.Uint16Size + 5 + msgp.Uint16Size + 9 + msgp.Uint32Size
+	s = 1 + 17
+	if z.MetricDefinition == nil {
+		s += msgp.NilSize
+	} else {
+		s += z.MetricDefinition.Msgsize()
+	}
+	s += 9 + msgp.Uint16Size + 6 + msgp.Uint16Size + 5 + msgp.Uint16Size + 9 + msgp.Uint32Size
+	return
+}
+
+// DecodeMsg implements msgp.Decodable
+func (z *Md5Hash) DecodeMsg(dc *msgp.Reader) (err error) {
+	var field []byte
+	_ = field
+	var zb0001 uint32
+	zb0001, err = dc.ReadMapHeader()
+	if err != nil {
+		err = msgp.WrapError(err)
+		return
+	}
+	for zb0001 > 0 {
+		zb0001--
+		field, err = dc.ReadMapKeyPtr()
+		if err != nil {
+			err = msgp.WrapError(err)
+			return
+		}
+		switch msgp.UnsafeString(field) {
+		case "Upper":
+			z.Upper, err = dc.ReadUint64()
+			if err != nil {
+				err = msgp.WrapError(err, "Upper")
+				return
+			}
+		case "Lower":
+			z.Lower, err = dc.ReadUint64()
+			if err != nil {
+				err = msgp.WrapError(err, "Lower")
+				return
+			}
+		default:
+			err = dc.Skip()
+			if err != nil {
+				err = msgp.WrapError(err)
+				return
+			}
+		}
+	}
+	return
+}
+
+// EncodeMsg implements msgp.Encodable
+func (z Md5Hash) EncodeMsg(en *msgp.Writer) (err error) {
```

Comment:
why do we need msgp support for this? it's only used in 1 in-memory-only datastructure, right?

Reply:
We definitely do not need msgp support for this. I will add an exclusion.